File: Java/leetcode75/src/main/java/org/phoenix/leetcode/challenges/Problem03_KidsWithTheGreatestNumberOfCandies.java
package org.phoenix.leetcode.challenges;

import java.util.ArrayList;
import java.util.List;

public class Problem03_KidsWithTheGreatestNumberOfCandies {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int max = 0;
        List<Boolean> resList = new ArrayList<>();
        for (int j : candies) {
            if (max < j) max = j;
        }
        for (int candy : candies) {
            if (candy + extraCandies >= max) {
                resList.add(true);
            } else {
                resList.add(false);
            }
        }
        return resList;
    }
}

File: JavaScript/leetcode75/src/main/Problem03_KidsWithTheGreatestNumberOfCandies.js
/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = function (candies, extraCandies) {
  const max = Math.max(...candies);
  let result = [];
  for (let candy of candies) {
    if (candy + extraCandies >= max) {
      result.push(true);
    } else {
      result.push(false);
    }
  }
  return result;
};

module.exports = kidsWithCandies;
